api/nakadi-event-bus-api.yaml

```diff
@@ -623,6 +623,68 @@ paths:
           schema:
             $ref: '#/definitions/Problem'
 
+  '/event-types/{name}/cursors':
+      post:
+        tags:
+          - unmanaged-api
+          - monitoring
+          - management-api
+        security:
+          - oauth2: ['nakadi.event_stream.read']
+        description: |
+          Transforms a list of Cursors with `shift` into a list without `shift`s. This is useful when there is the need
+          for randomly access events in the stream.
+
+        parameters:
+          - name: name
+            in: path
+            description: EventType name
+            type: string
+            required: true
+          - name: X-Flow-Id
+            in: header
+            description: |
+              The flow id of the request, which is written into the logs and passed to called services. Helpful
+              for operational troubleshooting and log analysis.
+            type: string
+          - name: cursors
+            in: body
+            description: |
+              Cursors indicating the positions to be calculated. Given a initial cursor, with partition and offset, it's
+              possible to obtain another cursor that is relatively forward or backward to it. When `shift` is positive,
+              Nakadi will respond with a cursor that in forward `shif` positions based on the initial `partition` and
+              `offset`. In case `shift` is negative, Nakadi will move the cursor backward.
+
+              Note: It's not currently possible to shift cursors based on time. It's only possible to shift cursors
+              based on the number of events to forward of backward given by `shift`.
+            required: true
+            schema:
+              type: array
+              items:
+                - $ref: '#/definitions/ShiftedCursor'
+        responses:
+          '200':
+            description: OK
+            schema:
+              type: array
```

Comment:
Please consider following the guideline to use objects for top level data structures (http://zalando.github.io/restful-api-guidelines/compatibility/Compatibility.html#must-always-return-json-objects-as-toplevel-data-structures-to-support-extensibility) to enable future compatible extensions. Even though it is sometimes hard to imagine the use cases, the rule is based on long term experience about managing API changes.